### ISSUE TITLE
system-filepath is apparently now deprecated

### ISSUE THREAD

**Issue by twhitehead** [Gabriel439/Haskell-DirStream-Library#6]
The [package page](https://hackage.haskell.org/package/system-filepath-0.4.13.4) for system-filepath says

> Deprecated in favor of filepath
> 
> Please see: https://plus.google.com/+MichaelSnoyman/posts/Ft5hnPqpgEx

**Comment by Gabriel439** [comment]
I'm aware of the deprecation status, but the library is still functional.

The main reason I'm still on the `system-filepath` library is that all the alternatives (except for `path`) are just type synonyms for either `String` or `ByteString`, and I prefer the `FilePath` type to be a separate type.

**Comment by sergv** [comment]
Maybe `dirstream` package can have a cabal file flag that switches underlying filepath library? This way either having a separate type for `FilePath` or not depending on a deprecated package will be possible, depending on the user's needs.

**Comment by Gabriel439** [comment]
Perhaps Haskell's new support for Backpack could solve this issue?

**Comment by sergv** [comment]
Yes, Backpack might help to resolve the issue. However, I believe there's simple solution. Recently I noticed that `path`'s type for paths is not a type synonym any more - https://hackage.haskell.org/package/path-0.6.1/docs/Path.html#t:Path. Perhaps it would be a good candidate now?

**Comment by Gabriel439** [comment]
The main issue with the `Path` type is the absence of an `IsString` instance, so you can't use string literals for paths
